runtime/lua/vim/fs.lua

```diff
@@ -76,8 +76,11 @@ end
 --- The search can be narrowed to find only files or or only directories by
 --- specifying {type} to be "file" or "directory", respectively.
 ---
----@param names (string|table) Names of the files and directories to find. Must
----             be base names, paths and globs are not supported.
+---@param names (string|table|fun(name: string): boolean) Names of the files
```

Comment:
I like the syntax you used here, but does EmmyLua understand this?

Reply:
lua-language-server should understand it: https://github.com/sumneko/lua-language-server/wiki/Annotations#documenting-types

> fun(PARAM: TYPE): RETURN_TYPE